client/app/scripts/superdesk-publish/filters/publish-filters.js

```diff
@@ -352,8 +360,116 @@ function PublishFiltersController($scope, filters, notify, modal) {
     fetchPublishFilters();
 }
 
+ProductionTestController.$inject = ['$scope', 'filters', 'notify', 'modal', '$location', '$window', '$rootScope'];
+function ProductionTestController($scope, filters, notify, modal, $location, $window, $rootScope) {
+    $scope.preview = null;
+    $scope.selected = {};
+    $scope.selectedItem = {};
+    var UP = -1,
+    DOWN = 1,
+    MOVES = {
+        38: UP,
+        40: DOWN
+    };
+
+    $scope.resultType = [
+        {id: 'Matching', value: 'true'},
+        {id: 'Non-Matching', value: 'false'}
+    ];
+
+    $scope.model = {selectedType:'true'};
+
+    $scope.close = function() {
+        $scope.filter_test = null;
+        $scope.testResult = null;
+    };
+    $scope.preview = function(Item) {
+        $location.search('_id', Item ? Item._id : Item);
+    };
+
+    $scope.$on('$routeUpdate', previewItem);
+
+    function previewItem() {
+        $scope.selectedItem = _.find($scope.testResult, {_id: $location.search()._id}) || null;
+        if ($scope.selectedItem) {
+            $scope.selected.preview = $scope.selectedItem;
+        } else {
+            $scope.selected.preview = null;
+        }
+    }
+    $scope.handleKeyEvent = function(event) {
+        var code = event.keyCode || event.which;
+        if (MOVES[code]) {
+            event.preventDefault();
+            event.stopPropagation();
+            move(MOVES[code], event);
+        }
+    };
+
+    function move(diff, event) {
+        var index = _.findIndex($scope.testResult, $scope.selectedItem),
+            nextItem,
+            nextIndex;
+
+        if (index === -1) {
+            nextItem = $scope.testResult[0];
+        } else {
+            nextIndex = Math.max(0, Math.min($scope.testResult.length - 1, index + diff));
+            nextItem = $scope.testResult[nextIndex];
+        }
+        clickItem($scope.testResult[nextIndex], event);
+    }
+    function select(item) {
+        $scope.selectedItem = item;
+        $location.search('_id', item ? item._id : item);
+    }
+
+    function clickItem(item, event) {
+        select(item);
+        if (event) {
+            event.preventDefault();
+            event.stopPropagation();
+            event.stopImmediatePropagation();
+        }
+    }
+    $scope.fetchResults = function() {
+        fetchProductionTestResult();
+    };
+
+    $scope.openViewTab = function(guid) {
+        var _url = '#/authoring/' + guid + '/view';
+        $window.open(_url, '_blank');
```

Comment:
Styling tip: no real need to use the underscore prefix here, it's obvious that `url` is local to the function.